src/WebJobs.Script/Binding/DocumentDBScriptBindingProvider.cs

```diff
@@ -89,25 +104,54 @@ public override Collection<Attribute> GetAttributes()
 
                 string databaseName = Context.GetMetadataValue<string>("databaseName");
                 string collectionName = Context.GetMetadataValue<string>("collectionName");
+                string connection = Context.GetMetadataValue<string>("connection");
 
-                DocumentDBAttribute attribute = null;
-                if (!string.IsNullOrEmpty(databaseName) || !string.IsNullOrEmpty(collectionName))
+                if (Context.IsTrigger)
                 {
-                    attribute = new DocumentDBAttribute(databaseName, collectionName);
+                    DocumentDBTriggerAttribute attributeTrigger = new DocumentDBTriggerAttribute(databaseName, collectionName);
+                    attributeTrigger.ConnectionString = connection;
+                    string leaseCollectionName = Context.GetMetadataValue<string>("leaseCollectionName");
+                    string leaseDatabaseName = Context.GetMetadataValue<string>("leaseDatabaseName");
+                    string leaseConnectionString = Context.GetMetadataValue<string>("leaseConnection");
+
+                    if (!string.IsNullOrEmpty(leaseCollectionName))
+                    {
+                        attributeTrigger.LeaseCollectionName = leaseCollectionName;
```

Comment:
how does this work if not all of these lease properties are set?

Also, it might be preferable to bundle all these lease properties in one object, i.e.

```
{
  "type": "documentDbTrigger",
  "connection": ...
  "lease": {
    "collection": ...,
    "database": ...,
    "connection": ...,
    "options": { ... }
  }
}